lua/cl_spectator_deathmatch.lua

```diff
@@ -493,15 +485,15 @@ net.Receive("SpecDM_Autoswitch", function()
 	end
 end)
 
-local hitmarker_enabled = false
+local hitmarker = CreateClientConVar("ttt_specdm_hitmarker", 1, FCVAR_ARCHIVE)
 local hitmarker_deadly = false
-local hitmarker = CreateClientConVar("ttt_specdm_hitmarker", "1", FCVAR_ARCHIVE)
 
 local respawntime = -2
 local autorespawntime = -2
 hook.Add("HUDPaint", "HUDPaint_SpecDM", function()
 	if !IsValid(LocalPlayer()) or !LocalPlayer():IsGhost() then return end
-	if hitmarker:GetBool() and hitmarker_enabled then
+	if hitmarker:GetBool() then
+		if hitmarker_remain and hitmarker_remain > CurTime() then return end
```

Comment:
Should be `if hitmarker_remain and hitmarker_remain < CurTime() then return end` otherwise it's always showing.